code/obj/machinery/phosphorizer.dm

```diff
@@ -0,0 +1,163 @@
+		var/colr2 = num2hex(round((510 + ctrl_R) / 3),2)
+		var/colg2 = num2hex(round((510 + ctrl_G) / 3),2)
+		var/colb2 = num2hex(round((510 + ctrl_B) / 3),2)
+		phos_target.color = "#[colr2][colg2][colb2]"
+
+		var/nameadjust = istype_exact(phos_target,/obj/item/light/tube) ? "tube" : "bulb"
+
+		phos_target.name = "phosphorized light [nameadjust]"
+		phos_target.desc = "A light [nameadjust] that has been coated with a phosphor to change its hue. A small label is marked '[ctrl_R]-[ctrl_G]-[ctrl_B]'."
+
+		phos_target.pixel_x = -4
+		phos_target.pixel_y = -8 //move bulb more towards output chute
+		phos_target.set_loc(src.loc)
+
+		playsound(src.loc, sound_process, 80, 1)
+
+	proc/stop_phos()
+		src.phosphorizing = false
+		UpdateOverlays(null, "operatebar", 0, 1)
+
+	proc/start_phos()
+		src.phosphorizing = true
+		var/image/O_panel = SafeGetOverlayImage("operatebar", 'icons/obj/machines/phosphorizer.dmi', "activelight")
+		UpdateOverlays(O_panel, "operatebar", 0, 1)
+		sleep(phos_delay)
+
+		for (var/obj/item/M in src.contents)
+			if(status & NOPOWER || src.phosphorizing == false)
+				failbreak = true
+				break
+			use_power(src.power_usage)
+			colorize_bulb(M)
+
+			sleep(phos_delay)
+
+		if(failbreak)
+			failbreak = false
+			src.visible_message("<b>[src]</b> abruptly stops operating.")
+			playsound(src.loc, sound_grump, 20, 1)
+		else
+			src.visible_message("<b>[src]</b> finishes working and shuts down.")
+			playsound(src.loc, sound_happi, 20, 1)
+		if(src.phosphorizing) stop_phos()
+
+/obj/machinery/phosphorizer/power_change()
+	var/image/I_panel = SafeGetOverlayImage("statuspanel", 'icons/obj/machines/phosphorizer.dmi', "powerpanel")
+	I_panel.plane = PLANE_SELFILLUM
+	I_panel.alpha = 128
+	if (status & BROKEN)
+		UpdateOverlays(null, "statuspanel", 0, 1)
+		if(src.phosphorizing) src.stop_phos()
+	else
+		if ( powered() )
+			UpdateOverlays(I_panel, "statuspanel", 0, 1)
+			status &= ~NOPOWER
+		else
+			SPAWN_DBG(rand(0, 5))
+				UpdateOverlays(null, "statuspanel", 0, 1)
+				status |= NOPOWER
+
+/obj/machinery/phosphorizer/ui_interact(mob/user, datum/tgui/ui)
+	ui = tgui_process.try_update_ui(user, src, ui)
+	if (!ui)
+		ui = new(user, src, "Phosphorizer", name)
+		ui.open()
+
+/obj/machinery/phosphorizer/ui_data(mob/user)
+	. = list(
+		"lights" = src.contents.len,
+		"online" = src.phosphorizing,
+		"hostR" = src.ctrl_R,
+		"hostG" = src.ctrl_G,
+		"hostB" = src.ctrl_B,
+	)
+
+/obj/machinery/phosphorizer/ui_act(action, list/params, datum/tgui/ui, datum/ui_state/state)
+	. = ..()
+	if (.)
+		return
+	switch(action)
+		if("tune_hue")
+			switch(params["hue"])
+				if("R") src.ctrl_R = clamp(round(params["output"]), 20, 255)
+				if("G") src.ctrl_G = clamp(round(params["output"]), 20, 255)
+				if("B") src.ctrl_B = clamp(round(params["output"]), 20, 255)
+			. = TRUE
+		if("toggle-process")
+			if(status | BROKEN && powered() && src.contents.len)
```

Comment:
```suggestion
			if (length(src.contents))
```
`ui_act` is bombs out earlier than this [here](https://github.com/goonstation/goonstation/blob/master/code/modules/tgui/external.dm#L78) if `ui.status != UI_INTERACTIVE`.

Machinery status is (by default) determined [here](https://github.com/goonstation/goonstation/blob/master/code/obj/machinery.dm#L186), which uses a "not broken" check that `/obj/machinery` has [here](https://github.com/goonstation/goonstation/blob/master/code/modules/tgui/states/not_broken.dm#L15). That part checks for whether it's broken, and whether (if it needs power) if it has power.

Basiclally: the checks here are superfluous, so I suggested them removed.

Reply:
Removal done in f7358dc.